src/instagram.erl

```diff
@@ -0,0 +1,314 @@
+-module(instagram).
+-behaviour(gen_oauth2).
+
+-include("gen_oauth2.hrl").
+
+-export([start/0, start/3, start/4, stop/1]).
+
+% Authorization
+-export([get_authorization_url/1,
+         get_access_token/1,
+         identity/1,
+         set_access_token/2]).
+% Users
+-export([get_user/1, get_user/2,
+         get_feed/1, get_feed/2,
+         get_recent_media/1, get_recent_media/2,
+         get_liked/1, get_liked/2,
+         search_users/2, search_users/3]).
+% Relationship
+-export([get_follows/1,
+         get_followed_by/1,
+         get_requested_by/1,
+         get_relationship/2]).
+% Media
+-export([get_media/2,
+         get_media_by_shortcut/2,
+         get_popular_media/1,
+         search_media/2]).
+% Comments
+-export([get_media_comments/2]).
+% Likes
+-export([get_media_likes/2]).
+% Tags
+-export([get_tag/2,
+         get_tag_recent_media/2, get_tag_recent_media/3,
+         search_tags/2]).
+
+-export([start_link/1, start_link/2, start_link/3]).
+
+-define(CLIENT_TOKEN_URL(ClientId, RedirectURI),
```

Comment:
I was under the impression that `lists:flatten()` is more efficient than `++`. Is this not the case for macros?